### ISSUE TITLE
Media Overlays reading order

### ISSUE THREAD

**Issue by rdeltour** [w3c/epub-specs#1458]
The MO spec currently says in [3.2.1 Structure](https://www.w3.org/publishing/epub32/epub-mediaoverlays.html#sec-media-overlays-structure).

> The ordering of [Media Overlays] elements MUST match the default reading order of the EPUB Content Document.

Can this conformance requirement be clarified? Should it be moved to the accessibility spec instead?

Someone from a DAISY member organization recently brought up to us an interesting use case: an old Chinese poem where the reading order differs from the visual order. In display, the text has three paragraphs A, B, C. The reading order is A, C, B.

The issue with the requirement above is that "reading order" isn't defined anywhere. 

As @mattgarrish said (in an internal discussion around this):
> Looking at it again, the wording is kind of ambiguous, as it doesn't say it must match the order of elements in the file like we had for the table of contents. I suppose you could read the requirement that if the default reading order is not sequential then the order of the media overlay also must be non-sequential
>(…)
> The accessibility requirement for elements to be ordered as you'd expect to read them is a content issue.

@marisademeglio then pointed out that this is somewhat linked to [WCAG SC 1.3.2](https://www.w3.org/TR/WCAG21/#meaningful-sequence) which says:

> When the sequence in which content is presented affects its meaning, a correct reading sequence can be programmatically determined

In my understanding, and to simply, the DOM order provides the reading order. For the use case above, a solution would be to have the DOM and MO respect the meaningful reading sequence, and to use CSS to visually re-order the paragraphs.

@marisademeglio says that footnotes are an interesting use case too. They can be at the bottom of a text document, but inline and skippable in the SMIL file.

I'm hoping we can clarify all this in 3.3.

**Comment by llemeurfr** [comment]
> For the use case above, a solution would be to have the DOM and MO respect the meaningful reading sequence, and to use CSS to visually re-order the paragraphs.

I doubt most content producers will see it this way; such solution would make ebook development more complicated. If we make the parallel with sync narration for comics, in complex cases, the layout of panels in an image is not re-arranged dynamically by CSS; on the contrary the sequence of image fragments which is associated with audio or text follows the "meaningful reading sequence" whatever the visual layout is.   

More, this conformance requirement is non properly testable IMO, therefore it should not be expressed like that. It seems that following a meaningful reading sequence is simply part of the definition of Media Overlays: do we even need to express this in specific terms?

**Comment by mattgarrish** [comment]
> I doubt most content producers will see it this way

It's not the producer's discretion to meet accessibility requirements in this case, though. Jumbling the reading order in the markup is one of the cardinal sins of accessibility, as this would also make the content unusable for screen readers. That's why I'm not sure if the requirement is unworkable or not.

Fixed layouts aren't a useful comparison, since they're generally not accessible no matter what you do, and I doubt would be an option here, either. Whether CSS would work well is also questionable, but at least when columns couldn't be produced the default would remain readable. (But we don't have to figure out the markup for this case, fortunately.)

> It seems that following a meaningful reading sequence is simply part of the definition of Media Overlays

We can't rely on common sense to meet an accessibility threshold, as intuitive as many things might seem. If you remove the requirement, for example, someone could modify their content and forget to update the media overlays and despite there being a discrepancy there would be no formal failure. We may not care in general if this happens, but we need to when we're stating accessibility conformance.

But as @rdeltour has already mentioned, this may better belong in the media overlays requirements of the accessibility specification.

**Comment by avneeshsingh** [comment]
The concept here is that a user must get the information in the same order no matter one is using a screen reader, readAloud (in-built TTS in reading system) or Media Overlays.

And reading order with screen reader and read aloud greatly depends on the DOM. So, it is little difficult to understand, what change do we want in EPUB accessibility.

**Comment by mattgarrish** [comment]
Closing this with #1558 and #1563

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-03-11-epub-a11y#section3) on 2021-03-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 3. Media Overlays section is updated 

_See github issue [#1458](https://github.com/w3c/epub-specs/issues/1458)._




**Avneesh Singh:** The requirement of logical reading order is removed from main EPUB 3.3 Media Overlays specs, and placed in EPUB Accessibility.

> *Avneesh Singh:* [https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-mo](https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-mo)

**Matt Garrish:** the issue is: does the specs have to be so strict on the reading order for MO?  
… there are situations where you can have multiple ways to read the same content  
… the solution is to remove the requirement of following the reading order  
… from the main specs  

**Avneesh Singh:** maybe we can add the requirement from MO to follow the reading order in the EPUB accessibility guidelines  

**Matt Garrish:** I think that in different cases you can have multiple ways to read the content  

**Avneesh Singh:** my question came from WCAG where following the reading order is a requirement  
… since MO is an alternative for the screenreader, maybe we have to think about this requirement  

**Matt Garrish:** yes, I understand  

**Bill Kasdorf:** my question is about skippability in publications that don't have a particular reading order, like reference and legal publications  
… if I look for a particular paragraph I want to get there  

</details>
